lib/Core/canBeDeleted.js

```diff
@@ -0,0 +1,10 @@
+function canBeDeleted(catalogMember) {
+    if(catalogMember.isUserSupplied && catalogMember.parent){
+      if (catalogMember.parent.name === catalogMember.terria.catalog.userAddedDataGroup.name) {
```

Comment:
No need to check the names, just check the reference directly.
```suggestion
      if (catalogMember.parent === catalogMember.terria.catalog.userAddedDataGroup) {
```